### PR TITLE
Support parallel preparation/commit/rollback in Consensus Commit

### PR DESCRIPTION
**Pull request by brfrn169**
This PR introduces the following parameters:
```
scalar.db.consensus_commit.parallel_preparation.enabled
scalar.db.consensus_commit.parallel_commit.enabled
scalar.db.consensus_commit.parallel_rollback.enabled
scalar.db.consensus_commit.parallel_executor_count
```

When we set `parallel_preparation.enabled` to `true`, the record preparation will be done in parallel. Also, when we the `parallel_commit.enabled` and `parallel_rollback.enabled` parameter to `true`, the record commit/rollback will be done in parallel. The `parallel_executor_count` determines the number of threads for the parallel execution. 

Please take a look!

### CODE REVIEW

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! Thank you!
Left one design discussion topic. PTAL!

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java [hunk 1]
```diff
@@ -66,9 +81,13 @@ private void prepareRecords(Snapshot snapshot)
...
     PartitionedMutations mutations = new PartitionedMutations(composer.get());
 
-    for (PartitionedMutations.Key key : mutations.getOrderedKeys()) {
-      storage.mutate(mutations.get(key));
+    ImmutableList<PartitionedMutations.Key> orderedKeys = mutations.getOrderedKeys();
+    List<ThrowableRunnable<ExecutionException>> tasks = new ArrayList<>(orderedKeys.size());
+    for (PartitionedMutations.Key key : orderedKeys) {
+      tasks.add(() -> storage.mutate(mutations.get(key)));
     }
+    ScalarDbUtils.executeTasks(
```
[the remark this answers is not in the document]
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java:89]: Thank you for taking a look at this and the suggestion!

That sounds like a good idea. I'll try to change in that way.

[the remark this answers is not in the document]
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java]: @feeblefakie I introduced `ParallelExecutor` in the latest commit. Please take a look!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Maybe, it's nice to have unit tests for ParallelExecutor in another PR?

**Comment by brfrn169** [comment]
> Maybe, it's nice to have unit tests for ParallelExecutor in another PR?

Ah yes, I forgot to add unit tests for ParallelExecutor. I'll add them in another PR. Thank you!

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**